windows/hid.c

```diff
@@ -441,12 +464,15 @@ struct hid_device_info HID_API_EXPORT * HID_API_CALL hid_enumerate(unsigned shor
 				cur_dev->path = NULL;
 
 			/* Serial Number */
-			wstr[0]= 0x0000;
-			res = HidD_GetSerialNumberString(write_handle, wstr, sizeof(wstr));
-			wstr[WSTR_LEN-1] = 0x0000;
-			if (res) {
-				cur_dev->serial_number = _wcsdup(wstr);
-			}
+			for (i=0; i < 5; ++i) {		   
+				res = HidD_GetSerialNumberString(write_handle, wstr, sizeof(wstr));
+				wstr[WSTR_LEN-1] = 0x0000;
+				if (res) {
+					cur_dev->serial_number = _wcsdup(wstr);
+				} else {
+                    Sleep(100);
```

Comment:
this chunk makes no sense 
1) not related to the PR topic - remove from this PR
2) what does it tries to fix?